File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/GridLayoutManager.java
package ch.rmy.android.http_shortcuts.utils;

import android.content.Context;
import android.util.DisplayMetrics;

public class GridLayoutManager extends android.support.v7.widget.GridLayoutManager {

    private boolean empty;

    public GridLayoutManager(Context context) {
        super(context, getNumberOfColumns(context));

        setSpanSizeLookup(new android.support.v7.widget.GridLayoutManager.SpanSizeLookup() {
            @Override
            public int getSpanSize(int position) {
                if (position == 0 && empty) {
                    return getSpanCount();
                } else {
                    return 1;
                }
            }
        });
    }

    private static int getNumberOfColumns(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        float dpWidth = displayMetrics.widthPixels / displayMetrics.density;
        return (int) (dpWidth / 90);
    }

    public void setEmpty(boolean empty) {
        this.empty = empty;
    }


}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/GridLayoutManager.kt
package ch.rmy.android.http_shortcuts.utils

import android.content.Context
import androidx.recyclerview.widget.GridLayoutManager

class GridLayoutManager(context: Context) : GridLayoutManager(context, getNumberOfColumns(context)) {

    private var empty: Boolean = false

    init {
        spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
            override fun getSpanSize(position: Int): Int =
                if (position == 0 && empty) spanCount else 1
        }
    }

    fun setEmpty(empty: Boolean) {
        this.empty = empty
    }

    companion object {

        fun getNumberOfColumns(context: Context): Int {
            val displayMetrics = context.resources.displayMetrics
            val dpWidth = displayMetrics.widthPixels / displayMetrics.density
            return (dpWidth / 90).toInt()
        }

    }

}
